src/interp/interp.cc

```diff
@@ -245,9 +245,17 @@ void Store::DeleteRoot(RootList::Index index) {
 }
 
 void Store::Collect() {
+  assert(gc_context_ == nullptr);
+
   size_t object_count = objects_.size();
-  marks_.resize(object_count);
-  std::fill(marks_.begin(), marks_.end(), false);
+
+  GCContext context;
+
+  context.call_depth = 0;
+  context.marks.resize(object_count);
+  std::fill(context.marks.begin(), context.marks.end(), false);
+
+  gc_context_ = &context;
```

Comment:
This line sets of my danger detector, even though its been set to null at the end.    I guess passing `gc_context_` explicitly to all the `Mark` would be messier?

Reply:
Does C++17 have some way to automatically unassigned a member at the end of a scope maybe?

Reply:
Perhaps I could pass a reference of the `gc_context_` pointer to the GCContext constructor, and its destructor resets its to nullptr. Or creating a special class for this purpose. Which one do you prefer?